source/target/freescale/rapid_iot/target_reset.c

```diff
@@ -0,0 +1,187 @@
+/**
+ * @file    target_reset.c
+ *
+ * DAPLink Interface Firmware
+ * Copyright (c) 2009-2016, ARM Limited, All Rights Reserved
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "target_reset.h"
+#include "swd_host.h"
+#include "info.h"
+#include "target_config.h"
+
+#define MDM_STATUS  0x01000000
+#define MDM_CTRL    0x01000004
+#define MDM_IDR     0x010000fc
+#define MDM_ID      0x001c0000 // K64, K22 (K series)
+
+/* Kinetis series ID */
+#define K_SERIES     0
+#define KW_SERIES    5
+
+/* KW4 subfamily defines */
+#define KW40         0
+#define KW41         1
+
+void target_before_init_debug(void)
+{
+    swd_set_target_reset(1);
+}
+
+void prerun_target_config(void)
+{
+    // SIM peripheral   0x40047000
+    // address offset   0x    1054
+    uint32_t UUID_LOC = 0x40048054;
+    uint32_t uuid[4] = {0};
+    // SIM peripheral   0x40047000
+    // address offset   0x    1024
+    uint32_t SDID = 0x40048024;
+    uint32_t sdid;
+
+    // get a hold of the target
+    target_set_state(RESET_PROGRAM);
```

Comment:
The target is getting reset on boot which could cause problems for persistent storage.

Reply:
This is there in the current code for Kinetis devices and not something that was added as part of this change.

Reply:
This was removed in #388. You'll need to pull in that same change here.